### ISSUE TITLE
Options for Autosuggest Support

### ISSUE THREAD

**Issue by maou-shonen** [elbywan/wretch#253]
Hi! I really like the API of Wretch,
but I encountered issue #251 right when I started using it.

It would be fantastic to have autosuggest support.
Being loosely typed shouldn’t prevent the addition of autosuggestions,
here’s a possible example.

```ts
type Options = Omit<
  RequestInit,
  "SomeWretchExcludedOptions1" | "SomeWretchExcludedOptions2"
> & {
  SomeWretchOnlyOption1: string
  SomeWretchOnlyOption2: number
  [otherKey: string]: any
}
```

**Comment by elbywan** [comment]
Hey @maou-shonen,

> Being loosely typed shouldn’t prevent the addition of autosuggestions

No you're right, but to be honest the loose typing is not the only reason why I typed the options this way.
It also caused some issues in the past with conflicting types: https://github.com/elbywan/wretch/issues/83

But I believe time has passed since then and I can try to reintroduce some form of type narrowing :+1: .

**Comment by elbywan** [comment]
:shipit: Released in [v2.11.0](https://github.com/elbywan/wretch/releases/tag/2.11.0)
